src/app/offer-investment-info/offer-investment-info.component.ts

```diff
@@ -0,0 +1,34 @@
+import {ChangeDetectionStrategy, Component, Input} from '@angular/core'
+import {CampaignWithInfo} from "../shared/services/blockchain/campaign.service"
+import {PercentPipe} from "@angular/common"
+
+@Component({
+  selector: 'app-offer-investment-info',
+  templateUrl: './offer-investment-info.component.html',
+  styleUrls: ['./offer-investment-info.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class OfferInvestmentInfoComponent {
+
+  @Input() display: 'narrow' | 'wide' | 'return-only' = 'narrow'
+  @Input() offer!: CampaignWithInfo
+
+  constructor(private percentPipe: PercentPipe,) {
+  }
+
+  hasReturnRate(): boolean {
+    return !!this.offer.return.from
+  }
+
+  returnRateRange(): string | null {
```

Comment:
can this return just `string`? I hope you set it here because transform might return `null`?

Reply:
```
      return this.percentPipe.transform(this.offer.return.from) || ''
```
Can you set this to avoid `null`?

Reply:
Yes, no problem 👍